src/main/java/service/InvoiceService.kt

```diff
@@ -40,4 +49,62 @@ class InvoiceService @Inject constructor(
 
         return invoiceDao.updateInvoice(invoice)
     }
+
+    fun generateVehiclesInvoices(country: Country, month: Date): List<Invoice> {
+        var invoices = arrayListOf<Invoice>()
+        val vehicles: List<Vehicle> = vehicleService.allVehicles()
+        val monthSeconds: Long = 60 * 60 * 24 * 30
+        val millisInSecond: Long = 1000
+        val expirationDate = Date(month.time + monthSeconds * millisInSecond)
+
+        vehicles.forEach {
+            val invoice = generateVehicleInvoice(it, country, month, expirationDate)
+            invoices.add(invoice)
+        }
+
+        return invoices
+    }
+
+    fun generateVehicleInvoice(vehicle: Vehicle, country: Country, month: Date, expirationDate: Date): Invoice {
+        var totalMeters = 0.0
+        val rider: Profile = vehicle.owner!!
+
+        vehicle.activities.filter {
+            it.creationDate.month == month.month
+            && it.creationDate.year == month.year
+            && it.country.name == country.name
+            && it.rider.id == rider.id
+        }.map {
+            totalMeters += getTotalDistanceOfActivity(it)
+        }
+
+        val invoice = Invoice(
+            InvoiceGenerationType.AUTO,
+            InvoiceState.OPEN,
+            rider,
+            vehicle,
+            expirationDate,
+            totalMeters / 1000
```

Comment:
Why not save the distance in meters and only convert it to KM when we report it?

Reply:
Changed to meters